src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,470 @@
+
+You can confirm that the node is still live and available using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+### 3. WAL Failover during long disk stalls
+
+#### Cause long disk stall
+
+Before triggering a WAL Failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that we can observe WAL Failover metrics.
+
+Next, we will trigger WAL Failover by introducing a transient disk stall that is `100s` in duration (longer than the value of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables)).
+
+Create a file named `long-stall.sh` with the following content which will cause long disk stalls:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+while (true)
+do
+    date
+    echo "Stalling for 100 seconds"
+    sudo echo '259:0 rbps=4 wbps=4' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 100
+    date
+    echo "Unstalling"
+    sudo echo '259:0 rbps=max wbps=max' > /sys/fs/cgroup/system.slice/cockroach-system.service/io.max
+    sleep 10
+    date
+    exit
+done
+~~~
+
+Next, make the script file executable:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+chmod 777 wal-flip.sh
+~~~
+
+Next, cause a long stall:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+long-stall.sh
+~~~
+
+#### WAL Failover metrics
+
+{% include {{ page.version.version }}/wal-failover-metrics.md %}
+
+Whenever a WAL failover occurs on a disk, `wal.failover.switch.count` for the associated store will increment by 1.
+
+When the disk continues to be stalled for longer than the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables), the node goes down, and there is no more metrics data coming from that node.
+
+You can confirm that the node is down (no longer live and available) using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+## Summary
+
+### Important environment variables
+
+Variable | Description | Default | Recommended value with WAL failover enabled
+---------|-------------|---------|---------------------------------------------
+`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT` | The threshold above which an observed engine sync duration triggers a fatal error. This environment variable is the default for the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting. If that cluster setting is explicitly set, it takes precedence over this environment variable. | `20s`   | `40s`
+`COCKROACH_LOG_MAX_SYNC_DURATION` | The maximum duration the file sink is allowed to take to write a log entry before we fatal the process for a disk stall. | `20s` | `40s`
```

Comment:
Not sure what "we fatal" means, but don't use "we" (per style) in any case.

```suggestion
`COCKROACH_LOG_MAX_SYNC_DURATION` | The maximum duration the file sink is allowed to take to write a log entry before we fatal the process for a disk stall. | `20s` | `40s`
```

Reply:
updated to "before the `cockroach` process is killed due to a disk stall"